SetVariablesTest.shouldFailSetVariablesIfWorkflowInstanceIsCompleted is flaky

```
16:10:39 [1 - Java Tests] [ERROR] Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.644 s <<< FAILURE! - in io.zeebe.broker.it.workflow.SetVariablesTest
16:10:39 [1 - Java Tests] [ERROR] shouldFailSetVariablesIfWorkflowInstanceIsCompleted(io.zeebe.broker.it.workflow.SetVariablesTest)  Time elapsed: 0.065 s  <<< FAILURE!
16:10:39 [1 - Java Tests] java.lang.AssertionError: Expected test to throw (an instance of io.zeebe.client.cmd.ClientStatusException and status code to be <NOT_FOUND>)
16:10:39 [1 - Java Tests] 	at org.junit.Assert.fail(Assert.java:88)
16:10:39 [1 - Java Tests] 	at org.junit.rules.ExpectedException.failDueToMissingException(ExpectedException.java:263)
16:10:39 [1 - Java Tests] 	at org.junit.rules.ExpectedException.access$200(ExpectedException.java:106)
16:10:39 [1 - Java Tests] 	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:245)
16:10:39 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
16:10:39 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
16:10:39 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
16:10:39 [1 - Java Tests] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
16:10:39 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
16:10:39 [1 - Java Tests] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
16:10:39 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
16:10:39 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
16:10:39 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:128)
16:10:39 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:27)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
16:10:39 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
16:10:39 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
16:10:39 [1 - Java Tests] 
```

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/workflow/SetVariablesTest.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/workflow/SetVariablesTest.java
@@ -51,9 +51,11 @@ public class SetVariablesTest {
           .endEvent("end")
           .done();
 
-  public static EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
-  public static GrpcClientRule clientRule = new GrpcClientRule(brokerRule);
-  @ClassRule public static RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(clientRule);
+  private static final EmbeddedBrokerRule BROKER_RULE = new EmbeddedBrokerRule();
+  private static final GrpcClientRule CLIENT_RULE = new GrpcClientRule(BROKER_RULE);
+
+  @ClassRule
+  public static final RuleChain RULE_CHAIN = RuleChain.outerRule(BROKER_RULE).around(CLIENT_RULE);
 
   @Rule
   public RecordingExporterTestWatcher recordingExporterTestWatcher =
@@ -64,14 +66,14 @@ public class SetVariablesTest {
   @BeforeClass
   public static void init() {
     final DeploymentEvent deploymentEvent =
-        clientRule
+        CLIENT_RULE
             .getClient()
             .newDeployCommand()
             .addWorkflowModel(WORKFLOW, "workflow.bpmn")
             .send()
             .join();
 
-    clientRule.waitUntilDeploymentIsDone(deploymentEvent.getKey());
+    CLIENT_RULE.waitUntilDeploymentIsDone(deploymentEvent.getKey());
   }
 
   @Test
@@ -80,7 +82,7 @@ public class SetVariablesTest {
     final long workflowInstanceKey = createWorkflowInstanceAndAwaitTaskActivation();
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables(VARIABLES)
@@ -100,7 +102,7 @@ public class SetVariablesTest {
     final long workflowInstanceKey = createWorkflowInstanceAndAwaitTaskActivation();
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables("null")
@@ -127,7 +129,7 @@ public class SetVariablesTest {
             "Property 'document' is invalid: Expected document to be a root level object, but was 'ARRAY'"));
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables("[]")
@@ -141,7 +143,7 @@ public class SetVariablesTest {
     final long workflowInstanceKey = createWorkflowInstanceAndAwaitTaskActivation();
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables(VARIABLES)
@@ -149,7 +151,7 @@ public class SetVariablesTest {
         .join();
     assertVariableDocumentUpdated();
 
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newWorker()
         .jobType("task-1")
@@ -170,7 +172,7 @@ public class SetVariablesTest {
     final long workflowInstanceKey = createWorkflowInstanceAndAwaitTaskActivation();
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables(Collections.singletonMap("foo", "bar"))
@@ -192,7 +194,7 @@ public class SetVariablesTest {
     newVariables.foo = "bar";
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables(newVariables)
@@ -210,7 +212,7 @@ public class SetVariablesTest {
   public void shouldFailSetVariablesIfWorkflowInstanceIsCompleted() {
     // given
     final long workflowInstanceKey = createWorkflowInstanceAndAwaitTaskActivation();
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newWorker()
         .jobType("task-1")
@@ -218,14 +220,14 @@ public class SetVariablesTest {
             (client, job) ->
                 client.newCompleteCommand(job.getKey()).variables("{\"result\": \"done\"}").send())
         .open();
-    assertWorkflowInstanceCompleted("process");
+    assertWorkflowInstanceCompleted("process", workflowInstanceKey);
 
     // then
     thrown.expect(ClientStatusException.class);
     thrown.expect(hasStatusCode(Code.NOT_FOUND));
 
     // when
-    clientRule
+    CLIENT_RULE
         .getClient()
         .newSetVariablesCommand(workflowInstanceKey)
         .variables(VARIABLES)
@@ -235,7 +237,7 @@ public class SetVariablesTest {
 
   private long createWorkflowInstanceAndAwaitTaskActivation() {
     final long workflowInstanceKey =
-        clientRule
+        CLIENT_RULE
             .getClient()
             .newCreateInstanceCommand()
             .bpmnProcessId("process")

